packages/clerk-js/src/ui/signUp/SignUpContinue.tsx

```diff
@@ -0,0 +1,350 @@
+import { Control } from '@clerk/shared/components/control';
+import { Form } from '@clerk/shared/components/form';
+import { Input } from '@clerk/shared/components/input';
+import { PhoneInput } from '@clerk/shared/components/phoneInput';
+import { SignUpResource } from '@clerk/types';
+import React from 'react';
+import type { FieldState } from 'ui/common';
+import {
+  buildRequest,
+  Footer,
+  handleError,
+  PoweredByClerk,
+  Separator,
+  useFieldState,
+  withRedirectToHome,
+} from 'ui/common';
+import { Body, Header } from 'ui/common/authForms';
+import { useCoreClerk, useCoreSignUp, useEnvironment, useSignUpContext } from 'ui/contexts';
+import { useNavigate } from 'ui/hooks';
+
+import { SignInLink } from './SignInLink';
+import { SignUpOAuth } from './SignUpOAuth';
+import { SignUpWeb3 } from './SignUpWeb3';
+import { determineFirstPartyFields } from './utils';
+
+type ActiveIdentifier = 'emailAddress' | 'phoneNumber';
+
+function _SignUpContinue(): JSX.Element | null {
+  const { navigate } = useNavigate();
+  const environment = useEnvironment();
+  const { displayConfig, userSettings } = environment;
+  const { setSession } = useCoreClerk();
+  const { navigateAfterSignUp } = useSignUpContext();
+  const [emailOrPhoneActive, setEmailOrPhoneActive] = React.useState<ActiveIdentifier>('emailAddress');
+  const signUp = useCoreSignUp();
+
+  // Redirect to sign-up if there is no persisted sign-up
+  if (!signUp.id) {
+    void navigate(displayConfig.signUpUrl);
+    return null;
+  }
+
+  // Pre-populate fields from existing sign-up object
+  const formFields = {
```

Comment:
🔧 A dedicated hook to prepare the data for the `<SignUpContinue/>` will be helpful as it will enable us to do more targeted unit testing. 

Moreover, it would also help us to move all the field creation logic to the SignUp resource so that the React components can be more dummy.

Reply:
I have moved the field determination logic to utils for the time being.

Don't know if SignUp would be the proper place, but maybe a form helper would be convenient.